### ISSUE TITLE
Terraform cloud : Setup failed: Failed to copy tfVars file: scp: /terraform/providers/github: No such file or directory

### ISSUE THREAD

**Issue by lemaadi** [hashicorp/terraform#27457]
Hello there,

I'm trying to upgrade the code from terraform version `0.11.14` to `0.12.0`. I'm using terraform cloud and remote as backend. 
The configuration is working correctly with terraform `0.11.14` version ( workspace is configured and linked with Github repository and terraform variables are set).

I just updated terraform version to `0.12.0` using terraform console and the working directory still unchanged ( `providers/github` )

### Steps before the upgrade

Run `terraform 0.12checklist` command : 

```
Looks good! We did not detect any problems that ought to be
addressed before upgrading to Terraform v0.12.

This tool is not perfect though, so please check the v0.12 upgrade
guide for additional guidance, and for next steps:
    https://www.terraform.io/upgrade-guides/0-12.html
```

### Upgrade

Run `terraform 0.12upgrade` command : 

```
This command will rewrite the configuration files in the given directory so
that they use the new syntax features from Terraform v0.12, and will identify
any constructs that may need to be adjusted for correct operation with
Terraform v0.12.

We recommend using this command in a clean version control work tree, so that
you can easily see the proposed changes as a diff against the latest commit.
If you have uncommited changes already present, we recommend aborting this
command and dealing with them before running this command again.

Would you like to upgrade the module in the current directory?
  Only 'yes' will be accepted to confirm.

  Enter a value: yes

-----------------------------------------------------------------------------

Upgrade complete!

The configuration files were upgraded successfully. Use your version control system to review the proposed changes, make any necessary adjustments, and then commit.
```

### Terraform Version
`0.12.0`

### Initialization

Run `terraform init` command : 

```
Initializing the backend...

Successfully configured the backend "remote"! Terraform will automatically
use this backend unless the backend configuration changes.

Initializing provider plugins...

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.

``` 

### Expected Behavior

The `terraform plan` command succeeds and shows the plan 

### Actual Behavior

Get this error : 

```
Waiting for the plan to start...

Terraform v0.12.0

Setup failed: Failed to copy tfVars file: scp: /terraform/providers/github: No such file or directory
```

### Steps to Reproduce

Please list the full steps required to reproduce the issue, for example:
1. `terraform init`
2. `terraform plan`

### Question 
Why terraform is pointing to this `/terraform/providers/github` path rather than `providers/github/` which configured in terraform workspace ?

**Comment by lemaadi** [comment]
Upgrading terraform to 0.12.30 fixed the issue

**Comment by dgreeninger** [comment]
Probably not related to this specific issue, but this was the first search result when I was investigating my `Setup failed: Failed to copy tfVars file` error message, so I'll share here to help out, if someone else happens across the same error.

My VCS link to terraform was malformed…  I had changed the hostname of my TFE from an IP to an actual DNS entry as recommended and it messed up the OAuth Application link to github. The repo list and the VCS link still worked in the workspace creation and I didn’t get any other obvious errors in the web UI, but automatic plan/applys weren’t working, when I pushed to Github and I was getting the `Setup failed: Failed to copy tfVars file: Failed copying file` error when I manually ran the workspace. 

After removing my VCS provider in the Settings on my organization, and re-creating it in Github with the new hostname and application name, everything went back to normal.
